pkg/storageincentives/agent.go

```diff
@@ -115,118 +114,59 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 
 	defer a.wg.Done()
 
-	var (
-		currentRound atomic.Uint64
-		phaseEvents  = newEvents()
-	)
-	// cancel all possible running phases
+	phaseEvents := newEvents()
 	defer phaseEvents.Close()
 
-	commitF := func(ctx context.Context) {
-		phaseEvents.Cancel(claim)
-
-		round := currentRound.Load()
-
-		// the sample has to come from previous round to be able to commit it
-		sample, err := getSample(a.state.stateStore, round-1)
+	logPhaseResult := func(phase PhaseType, round uint64, err error, isPhasePlayed bool) {
 		if err != nil {
-			if errors.Is(err, storage.ErrNotFound) {
-				// In absence of sample, phase is skipped
-				return
-			}
-
-			a.logger.Error(err, "getSample for commit phase")
-			return
-		}
-
-		err = a.commit(ctx, sample, round)
-		if err != nil {
-			a.logger.Error(err, "commit")
+			a.logger.Error(err, "phase failed", "phase", phase, "round", round)
+		} else if isPhasePlayed {
```

Comment:
replace with isRoundPlayed

Reply:
Why to use `round` here? `isPhasePlayed` has more sense because it tells if phase of current round has played or not.
But it should be corrected to `hasPhasePlayed`, do you agree?